gibbon-rts/rts-c/gibbon_rts.c

```diff
@@ -1796,13 +1796,19 @@ void gib_show_usage(char** argv)
     printf("\n");
     printf("This binary was generated by the Gibbon compiler.\n");
     printf("\n");
-    printf("Usage: %s [OPTS] [size] [iters]\n", argv[0]);
+    printf("Usage: %s [OPTIONS...]\n", argv[0]);
 
     printf("\n");
     printf("Options:\n");
-    printf(" --buffer-size <bytes>      Set the buffer size (default %" PRId64 ").\n", gib_global_biginf_init_chunk_size);
-    printf(" --bench-input <path>       Set the input file read for benchmarking. Applies only\n");
-    printf("                            IF the program was *compiled* with --bench-fun. \n");
+    printf(" --biginf-buffer-size <bytes>   Set the buffer size (default %" PRId64 ").\n", gib_global_biginf_init_chunk_size);
+    printf(" --inf-buffer-size <bytes>      Set the buffer size (default %" PRId64 ").\n", gib_global_inf_init_chunk_size);
+    printf(" --bench-input <path>           Set the input file read for benchmarking. Applies only\n");
+    printf("                                If the program was *compiled* with --bench-fun. \n");
+    printf("\n");
+    printf(" --array-input <path>           Set the file from which to read the array input.\n");
+    printf(" --array-input-length <int>     Set the size of the array input file.\n");
+    printf(" --iterate <int>                Set the number of timing iterations to perform (default 1).\n");
+    printf(" --size-param <int>             A parameter for size available as a language primitive which allows user to specify the size at runtime (default 1).\n");
```

Comment:
I can't say I understand this line, but maybe I just need to read more...

Reply:
I am not 100% sure how to phrase this. 
I see that SizeParam in a primitve in the gibbon std-lib. The compiler generates code to create a call to get this variable which is set as a runtime arg.

Reply:
Maybe you have a suggestion to better phrase this?

Reply:
Not at this point, I'm afraid. I'll have to dive deeper... For now, if no one objects, you could probably merge it as is.

Reply:
sounds good to me. I can add a TODO here to recheck this later.